### PR TITLE
QgsSvgCache::svgAsPicture(): make sure the returned picture is not shared (fixes #18996)

### PR DESCRIPTION
**Pull request by rouault**
For some reason QPicture.detach() doesn't seem to always work as intented, at
least with QT 5.5 on Ubuntu 16.04
Serialization/deserialization is a safe way to be ensured we don't
share a copy.

Relates to a6eea7205c72a1be837ab43b79aad0c67a92a9b2

### CODE REVIEW

**Comment by rldhont** [comment]
Hi @rouault do you think, it's something to backport to release-2_18 ?

**Comment by rouault** [comment]
> Hi @rouault do you think, it's something to backport to release-2_18 ?

Wouldn't hurt probably, but let's see what @nyalldawson thinks about this.

**Comment by rouault** [comment]
@rldhont  Actually when looking at https://github.com/qgis/QGIS/blob/release-2_18/src/core/symbology-ng/qgsfillsymbollayerv2.cpp, it looks like https://github.com/qgis/QGIS/commit/a6eea7205c72a1be837ab43b79aad0c67a92a9b2 which incorporate the first set of thread-safety fix hadn't been backported, so that would be a prior condition I guess.

**Review by nyalldawson — APPROVED** [review]
Good catch!

File: src/core/symbology/qgssvgcache.cpp [hunk 1]
```diff
@@ -182,8 +182,12 @@ QPicture QgsSvgCache::svgAsPicture( const QString &path, double size, const QCol
     trimToMaximumSize();
   }
 
-  QPicture p = *( currentEntry->picture );
-  p.detach();
+  QPicture p;
+  // For some reason p.detach() doesn't seem to always work as intented, at
```
Comment (nyalldawson) [inline src/core/symbology/qgssvgcache.cpp]: "Intended"
